File: Algorithms/BinarySearch/BinarySearch.cpp
#include <algorithm>
#include <cassert>
#include <iostream>
#include <vector>

int getPosition(const std::vector<int> &numbers, int number) {
    size_t low = 0;
    size_t high = numbers.size() - 1;

    while (low <= high) {
        size_t mid = low + (high - low) / 2;

        if (numbers[mid] == number) {
            return mid + 1;
        }

        if (numbers[mid] < number) {
            low = mid + 1;
        } else {
            high = mid - 1;
        }
    }

    return -1;
}

int main() {
    size_t numberCount;
    std::cin >> numberCount;
    std::vector<int> numbers(numberCount);
    for (auto &number:numbers) {
        std::cin >> number;
    }

    assert(std::is_sorted(numbers.begin(), numbers.end()));

    size_t queryCount;
    std::cin >> queryCount;
    while (queryCount-- > 0) {
        int number;
        std::cin >> number;
        std::cout << getPosition(numbers, number) << " ";
    }
    std::cout << std::endl;

    return 0;
}

File: Algorithms/BinarySearch/BinarySearch.cs
namespace BinarySearch
{
    public static class BinarySearch
    {
        public static (int, List<int>) ParseLine(this string input)
        {
            List<int> list = new();
            
            if (!String.IsNullOrEmpty(input))
            {
                string[] chars = input.Split(' ');
                
                for(int i = 1; i < chars.Length; i++)
                {
                    list.Add(int.Parse(chars[i]));
                }
            }

            return (list.Count, list);
        }

        public static int GetPosition(List<int> numbers, int number)
        {
            int low = 0;
            int high = numbers.Count - 1;

            while (low <= high)
            {
                int mid = low + (high - low) / 2;

                if (numbers[mid] == number)
                {
                    return mid + 1;
                }

                if (numbers[mid] < number)
                {
                    low = mid + 1;
                }
                else
                {
                    high = mid - 1;
                }
            }

            return -1;
        }

        public static void Main()
        {
            string? input = Console.ReadLine();
            (int numbersCount, List<int> numbers) = input.ParseLine();
            
            numbers.Sort();

            input = Console.ReadLine();
            (int queryCount, List<int> query) = input.ParseLine();

            query.ForEach(x => Console.Write("{0} ", GetPosition(numbers, x)));
        }
    }
}